momento-sdk/src/main/java/momento/sdk/messages/CacheDictionaryGetFieldsResponse.java

```diff
@@ -0,0 +1,174 @@
+package momento.sdk.messages;
+
+import com.google.protobuf.ByteString;
+import grpc.cache_client.ECacheResult;
+import grpc.cache_client._DictionaryGetResponse;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import momento.sdk.exceptions.CacheServiceExceptionMapper;
+import momento.sdk.exceptions.SdkException;
+import momento.sdk.internal.StringHelpers;
+
+/** Response for a dictionary get fields operation */
+public interface CacheDictionaryGetFieldsResponse {
+  /**
+   * A successful dictionary get fields operation that found the keys with values in the dictionary.
+   */
+  class Hit implements CacheDictionaryGetFieldsResponse {
+    public final List<CacheDictionaryGetFieldResponse> responsesList = new ArrayList<>();
+
+    /**
+     * Constructs a dictionary get fields hit with a list of encoded keys and values.
+     *
+     * @param responses the retrieved dictionary.
+     */
+    public Hit(
+        List<ByteString> fields,
+        List<_DictionaryGetResponse._DictionaryGetResponsePart> responses) {
+
+      int counter = 0;
+      for (_DictionaryGetResponse._DictionaryGetResponsePart element : responses) {
+        if (element.getResult() == ECacheResult.Hit) {
+          responsesList.add(
+              new CacheDictionaryGetFieldResponse.Hit(fields.get(counter), element.getCacheBody()));
+        } else if (element.getResult() == ECacheResult.Miss) {
+          responsesList.add(new CacheDictionaryGetFieldResponse.Miss(fields.get(counter)));
+        } else {
+          responsesList.add(
+              new CacheDictionaryGetFieldResponse.Error(
+                  CacheServiceExceptionMapper.convert(
+                      new Exception(element.getResult().toString())),
+                  fields.get(counter)));
+        }
+        counter++;
+      }
+    }
+
+    /**
+     * Gets the retrieved dictionary of string keys and string values.
+     *
+     * @return the dictionary.
+     */
+    public Map<String, String> valueDictionaryStringString() {
```

Comment:
can we add an alias for this one that is just named `valueDictionary`?

Reply:
Will handle this in a separate PR for all the apis that use this type of accessors.